app/src/main/java/com/arygm/quickfix/ui/dashboard/AnnouncementsWidget.kt

```diff
@@ -0,0 +1,245 @@
+                    fontSize = 19.sp,
+                    modifier = Modifier.testTag("AnnouncementsTitle"))
+                TextButton(
+                    onClick = { showAll = !showAll },
+                    modifier = Modifier.testTag("ShowAllButton")) {
+                      Text(
+                          text = if (showAll) "Show Less" else "Show All",
+                          color = MaterialTheme.colorScheme.onSurface,
+                          style = MaterialTheme.typography.bodyMedium,
+                          fontWeight = FontWeight.SemiBold)
+                    }
+              }
+
+          Divider(
+              color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.2f),
+              thickness = 1.dp,
+              modifier = Modifier.testTag("AnnouncementsDivider"))
+
+          val itemsToShow = if (showAll) announcements else announcements.take(itemsToShowDefault)
+          itemsToShow.forEachIndexed { index, announcement ->
+            val pairs = imagesForAnnouncements[announcement.announcementId] ?: emptyList()
+            val bitmapToDisplay = pairs.firstOrNull()?.second
+            var category by remember { mutableStateOf(Category()) }
+            LaunchedEffect(Unit) {
+              categoryViewModel.getCategoryBySubcategoryId(
+                  announcement.category,
+                  onSuccess = {
+                    if (it != null) {
+                      category = it
+                    }
+                  })
+            }
+
+            AnnouncementItem(
+                announcement = announcement,
+                announcementImage = bitmapToDisplay,
+                categoryIcon = getCategoryIcon(category),
+                onClick = {
+                  announcementViewModel.selectAnnouncement(announcement)
+                  navigationActions.navigateTo(UserScreen.ANNOUNCEMENT_DETAIL)
+                })
+
+            if (index < itemsToShow.size - 1) {
+              Divider(
+                  color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.2f),
+                  thickness = 1.dp,
+                  modifier = Modifier.testTag("AnnouncementDivider_$index"))
+            }
+          }
+        }
+  }
+}
+
+@Composable
+fun AnnouncementItem(
+    announcement: Announcement,
+    announcementImage: Bitmap?,
+    categoryIcon: ImageVector,
+    onClick: () -> Unit
+) {
+
+  Row(
+      modifier =
+          Modifier.fillMaxWidth()
+              .padding(8.dp)
+              .clickable { onClick() }
+              .testTag("AnnouncementItem_${announcement.announcementId}"),
+      verticalAlignment = Alignment.Top) {
+        // Image
+        if (announcementImage != null) {
+          Image(
+              bitmap = announcementImage.asImageBitmap(),
+              contentDescription = "Announcement Image",
+              contentScale = ContentScale.Crop,
+              modifier =
+                  Modifier.size(40.dp)
```

Comment:
You should rather use relative values for the dp's bigger than 16

Reply:
This will be done in a future pr because it should also be done on the messagesWidget.